src/bitmaps.rs

```diff
@@ -0,0 +1,199 @@
+use std::borrow::Borrow;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::Path;
+use std::{fs, io};
+
+use endio_bit::BitReader;
+
+use allsorts::binary::read::ReadScope;
+use allsorts::bitmap::{self, BitDepth, BitmapSize, CBDTTable, CBLCTable, GlyphBitmapData};
+use allsorts::fontfile::FontFile;
+use allsorts::tables::FontTableProvider;
+use allsorts::tag::{self};
+
+use crate::cli::BitmapOpts;
+use crate::BoxError;
+
+pub fn main(opts: BitmapOpts) -> Result<i32, BoxError> {
+    let buffer = std::fs::read(&opts.font)?;
+    let scope = ReadScope::new(&buffer);
+    let font_file = scope.read::<FontFile>()?;
+    let table_provider = font_file.table_provider(opts.index)?;
+
+    let table = table_provider
+        .table_data(tag::CBLC)?
+        .or(table_provider.table_data(tag::EBLC)?)
```

Comment:
Should we use `or_else` for lazy evaluation? Ditto with the `EBDT` table below.

Reply:
Good idea. I tried applying your suggestion but the `Result<Option<_>>` returned by `table_data` complicates matters. If I wrap it in a closure for `or_else` the `?` to unwrap the `Result` to get at the Option no longer works. Ultimately, `table_data` is relatively cheap so I think I'll leave it as is since it reads better.